src/data.rs

```diff
@@ -61,8 +61,8 @@ where
     ) -> BoxedFuture<'a, Result<Self::Asset, Self::Error>> {
         Box::pin(async move {
             let mut bytes = Vec::new();
-            reader.read_to_end(&mut bytes).await?;
-            let custom_asset = serde_json::from_slice::<A>(&bytes)?;
+            reader.read_to_end(&mut bytes).await.unwrap();
+            let custom_asset = serde_json::from_slice::<A>(&bytes).unwrap();
```

Comment:
I don't think `unwrap` is okay here, we don't want to panic and crash the application if there is a malformed JSON.

Probably you need to map the error to make `Self::Error`